lib/src/node/value/calculation.dart

```diff
@@ -0,0 +1,134 @@
+void assertUnquotedStringOrInterpolation(Object arg) {
+  if ((arg is! SassString && arg is! CalculationInterpolation) ||
+      (arg is SassString && arg.hasQuotes)) {
+    jsThrow(JsError(
+        'Argument must be an unquoted SassString or CalculationInterpolation'));
+  }
+}
+
+/// The JavaScript `SassCalculation` class.
+final JSClass calculationClass = () {
+  var jsClass =
+      createJSClass('sass.SassCalculation', (Object self, [Object? _]) {
+    jsThrow(JsError("new sass.SassCalculation() isn't allowed"));
+  });
+
+  jsClass.defineStaticMethods({
+    'calc': (Object argument) {
+      assertCalculationValue(argument);
+      return SassCalculation.unsimplified('calc', [argument]);
+    },
+    'min': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(assertCalculationValue);
+      return SassCalculation.unsimplified('min', argList);
+    },
+    'max': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(assertCalculationValue);
+      return SassCalculation.unsimplified('max', argList);
+    },
+    'clamp': (Object min, [Object? value, Object? max]) {
+      if (value == null) {
+        assertUnquotedStringOrInterpolation(min);
+      } else if (max == null) {
+        assertUnquotedStringOrInterpolation(min);
+        assertUnquotedStringOrInterpolation(value);
+      }
+      [min, value, max].whereNotNull().forEach(assertCalculationValue);
+      return SassCalculation.unsimplified(
+          'clamp', [min, value, max].whereNotNull());
+    }
+  });
+
+  jsClass.defineMethods({
+    'assertCalculation': (SassCalculation self, [String? name]) => self,
+  });
+
+  jsClass.defineGetters({
+    // The `name` getter is included by default by `createJSClass`
+    'arguments': (SassCalculation self) => ImmutableList(self.arguments),
+  });
+
+  getJSClass(SassCalculation.unsimplified('calc', [SassNumber(1)]))
+      .injectSuperclass(jsClass);
+  return jsClass;
+}();
+
+/// The JavaScript CalculationOperation class
+final JSClass calculationOperationClass = () {
+  var jsClass = createJSClass('sass.CalculationOperation',
+      (Object self, String strOperator, Object left, Object right) {
+    var operator = CalculationOperator.values
+        .firstWhereOrNull((value) => value.operator == strOperator);
+    if (operator == null) {
+      jsThrow(JsError('Invalid operator: $strOperator'));
+    }
+    assertCalculationValue(left, checkUnquoted: false);
+    assertCalculationValue(right, checkUnquoted: false);
+    return SassCalculation.operateInternal(operator, left, right,
+        inMinMax: false, simplify: false);
+  });
+
+  jsClass.defineMethods({
+    'equals': (CalculationOperation self, Object other) => self == other,
+    'hashCode': (CalculationOperation self) => self.hashCode,
+  });
+
+  jsClass.defineGetters({
+    'operator': (CalculationOperation self) => self.operator.operator,
```

Comment:
Shouldn't this expose `left` and `right` as well?

Reply:
It seems like normal class properties are transparently available in JS (the tests verify this). I can implement them as getters manually if you prefer to call them out explicitly

Reply:
I think it's a safer bet to be explicit, yeah. I don't think dart2js provides strong stability guarantees for the JS API it generates.